content/fast/preload-critical-assets/codelab-preload-web-fonts.md

```diff
@@ -0,0 +1,62 @@
+---
+page_type: glitch
+title: Preload web fonts to improve loading speed
+author: garima
+description: |
+  In this codelab, learn how to improve the performance of a page by preloading
+  web fonts.
+web_updated_on: 2018-12-06
+web_published_on: 2018-11-05
+glitch: web-dev-preload-webfont
+---
+
+This code lab shows you how to preload web fonts using rel="preload" to remove any flash of unstyled text (FOUT).
+
+
+## Measure
+
+First measure how the website performs before adding any optimizations.
+Firstly, measure when fonts are loaded on your site. Follow these steps to check.
+1. Click on the Show Live button to view the live version of the your Glitch.
+2. Open DevTools by pressing CMD + OPTION + i / CTRL + SHIFT + i.
+3. Click on the Audits panel.
+4. Select the Performance checkbox.
+5. Click Run Audits to generate a report.
+
+The Lighthouse report that is generated  will show you  the fetching sequence of resources under **Maximum critical path latency**.
+
+<img class="screenshot" src="./lighthouse-before.png" alt="Lighthouse audit - Before Preload">
+
+In the above audit the web fonts are part of the critical request chain and fetched last. The [**critical request chain**](https://developers.google.com/web/tools/lighthouse/audits/critical-request-chains) represents the order of resources that are prioritized and fetched by the browser. In this application, the web fonts (Pacfico and Pacifico-Bold) are defined using the [@font-face](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/webfont-optimization#defining_a_font_family_with_font-face) rule and are the last resource fetched by the browser in the critical request chain. Typically, webfonts are lazy loaded which means that they are not loaded until the critical resources are downloaded (CSS, JS).
+
+Here is the sequence of the resources fetched in the application:
+
+<img class="screenshot" src="./network-request-before.png" alt="Network Request before preload">
+ 
+## Preloading Web fonts.
+In order to avoid FOUT, you can preload web fonts that are required immediately. Add the Link element for this application at the head of the document:
+
+<pre class="prettyprint">
+&lt;head&gt;
+ &lt;!-- ... -->
+ <strong>&lt;link rel=&quot;preload&quot; href=&quot;/assets/Pacifico-Bold.woff2&quot; as=&quot;font&quot; type=&quot;font/woff2&quot; crossorigin&gt;</strong>
+&lt;/head&gt;
+</pre>
+
+The `as="font"` `type="font/woff2"` tells the browser to download this resource as a font and helps in prioritization of the re­source queue. 
+
+The `crossorigin` attribute indicates whether the resource should be fetched with a CORS request as the font may come from a different domain. Without this attribute, the preloaded font is ignored by the browser.
+
+Since Pacifico-Bold is used in page header, we added a preload tag to fetch it even sooner. It isn’t important to preload the Pacifico.woff2 font because it styles the text that is below the fold. 
```

Comment:
```suggestion
Since Pacifico-Bold is used in the page header, we added a preload tag to fetch it even sooner. It isn’t important to preload the Pacifico.woff2 font because it styles the text that is below the fold. 
```